spec2sdk/client/templates/client.j2

```diff
@@ -25,6 +25,20 @@ class APIClient:
     def __init__(self, *, http_client: HTTPClientProtocol):
         self._http_client = http_client
 
+    def serialize[T](self, instance: T, instance_type: Type[T] | TypeAliasType, content_type: str | None) -> bytes:
+        match content_type:
+            case "application/json":
+                return TypeAdapter(instance_type).dump_json(instance, by_alias=True)
+            case _:
+                return instance
+
+    def deserialize[T](self, response: bytes | None, response_type: Type[T] | TypeAliasType, content_type: str | None) -> T:
+        match content_type:
+            case "application/json":
+                return TypeAdapter(response_type).validate_json(response)
+            case _:
```

Comment:
Should we have an explicit `case` for `application/octet-stream`, e.g. something like
```suggestion
            case _:
```
or do you want it to be handled implicitly as the current code does it?

Reply:
There could be several options for other content types as well. For example, we can also have special handling for `text/plain` and `data_type=str` to convert bytes to string. Let's discuss it first. I'll keep current code for now.